lib/deliveries_manager/lib/deliveries_manager.rb

```diff
@@ -3,5 +3,24 @@
 require 'deliveries_manager/engine'
 
 module DeliveriesManager
-  # Your code goes here...
+  def self.processPayload(payload)
+    orderPayload = JSON.parse(payload)
+
+    if (!OrderProcessing.exists?(:order_id => orderPayload["order_id"]))
+      savedOP = OrderProcessing.new(orderPayload)
+      savedOP.save()
+    else
+      savedOP = OrderProcessing.where(:order_id => orderPayload["order_id"]).first(1)[0]
```

Comment:
why did you use [0]?

Reply:
first returns array, I needed the first item in the array.